drivers/serial/uart_sam0.c

```diff
@@ -377,6 +377,90 @@ static void uart_sam0_rx_timeout(struct k_work *work)
 
 #endif
 
+static int uart_sam0_configure(struct device *dev,
+								const struct uart_config *new_cfg)
+{
+	int retval;
+	const struct uart_sam0_dev_cfg *const cfg = DEV_CFG(dev);
+	SercomUsart * const usart = cfg->regs;
+
+	wait_synchronization(usart);
+
+	usart->CTRLA.bit.ENABLE = 0;
+	wait_synchronization(usart);
+
+	if (new_cfg->flow_ctrl != UART_CFG_FLOW_CTRL_NONE) {
+		/* Flow control not yet supported though in principle possible on this
+		 * soc family.
+		 */
+		return -ENOTSUP;
+	}
+
+	SERCOM_USART_CTRLA_Type CTRLA_temp = usart->CTRLA;
+	SERCOM_USART_CTRLB_Type CTRLB_temp = usart->CTRLB;
+
+	switch (new_cfg->parity) {
+	case UART_CFG_PARITY_NONE:
+		CTRLA_temp.bit.FORM &= 0xe;
```

Comment:
Since we do not support "auto-baud", we should just set `FORM` to 0; in case, the SERCOM was previously configured (e.g. by bootloader).

Reply:
I've mostly wanted to preserve any config the user might have done previously. Though, I thought any leftover bootloader settings would be overwritten [in the init function](https://github.com/zephyrproject-rtos/zephyr/blob/c25357a24764873760622ef96cfe678907d3f3e1/drivers/serial/uart_sam0.c#L491) anyway?

Reply:
> bootloader settings would be overwritten [in the init function](https://github.com/zephyrproject-rtos/zephyr/blob/c25357a24764873760622ef96cfe678907d3f3e1/drivers/serial/uart_sam0.c#L491) anyway?

You are right. Bootloader setting will not be a problem then.

I see another issue with this approach though, that LSB being set does not necessarily mean parity being enabled (although this pattern does seem to apply to the SAM D2x, even though there is no explicit mention of such in the datasheet).

For instance, refer to the `FORM` values for SAM D5x below:
![image](https://user-images.githubusercontent.com/11345418/76089432-fed52400-5ffc-11ea-9552-c0d1d868cdd8.png)

I wonder if we should handle this in a more verbose way?

Reply:
Good catch, I've only based this change on the samd2x datasheet. Even though the ISO7816 mode is not really used in the driver (and neither is the auto-baud one) it might indeed make sense to handle this in a more verbose way for the sake of future proofing it.